.github/workflows/rit.yml

```
@@ -23,29 +21,124 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
       - name: Set Branch Variables
         id: set-branch-variables
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          github_event_input_powpeg_branch: ${{ github.event.inputs.powpeg-branch }}
+          github_event_input_rit_branch: ${{ github.event.inputs.rit-branch }}
+          github_event_name: ${{ github.event_name }}
+          github_event_pull_request_number: ${{ github.event.pull_request.number }}
+          github_head_ref: ${{ github.head_ref }}
+          github_ref_name: ${{ github.ref_name }}
         run: |
-          # Default values
-          RSKJ_BRANCH="master"
-          RIT_BRANCH="${{ github.event.inputs.rit-branch || 'main' }}"
-          POWPEG_BRANCH="${{ github.event.inputs.powpeg-branch || 'master' }}"
-
-          if [[ "${{ github.event_name }}" == "push" && "${{ github.ref }}" == "refs/heads/*-rc" ]]; then
-            RSKJ_BRANCH="${{ github.ref }}"
-          elif [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
-            RSKJ_BRANCH="${{ github.ref_name }}"
-          elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
-            RSKJ_BRANCH="${{ github.head_ref }}"
+          PR_DESCRIPTION=pr-description.txt
+
+          ALLOWED_BRANCH_CHARACTERS='[-./0-9A-Z_a-z]'
+
+          default_rskj_branch=master
+          default_powpeg_branch=master
+          default_rit_branch=main
+
+          get_branch_from_description()
+          {
+            _prefix=$1
+
+            # On lines matching "`$_prefix:...`", replace the lines with the
+            # thing in ... and print the result.
+            _branch=$(sed -n "/\`$_prefix:$ALLOWED_BRANCH_CHARACTERS\{1,\}\`/ s/.*\`$_prefix:\($ALLOWED_BRANCH_CHARACTERS\{1,\}\)\`.*/\1/p" "$PR_DESCRIPTION")
```

Although this worked, it's unclear whether the `/` inside `$ALLOWED_BRANCH_CHARACTERS` should terminate the regex or not (it's mandated by POSIX.1-2024 but undefined by POSIX.1-2017 and previous). I prefer to err on the side of caution and use an alternative separator, `,`.

```suggestion
            _branch=$(sed -n "\\,\`$_prefix:$ALLOWED_BRANCH_CHARACTERS\{1,\}\`, s,.*\`$_prefix:\($ALLOWED_BRANCH_CHARACTERS\{1,\}\)\`.*,\1,p" "$PR_DESCRIPTION")
```